source/extensions/omni.isaac.lab/omni/isaac/lab/envs/mdp/actions/binary_joint_actions.py

```diff
@@ -115,6 +123,13 @@ def process_actions(self, actions: torch.Tensor):
             binary_mask = actions < 0
         # compute the command
         self._processed_actions = torch.where(binary_mask, self._close_command, self._open_command)
+        # clip actions
+        if self._clip is not None:
+            # resolve the dictionary config
```

Comment:
is it possible to resolve this once at initialization and convert to a torch tensor to avoid more overhead with the loop?

Reply:
OK, but I encountered a problem with the code

```python
# parse clip
        if cfg.clip is not None:
            if isinstance(cfg.clip, dict):
                print(cfg.clip)
                self._clip = torch.tensor([[-float('inf'), float('inf')]], device=self.device).expand(self.num_envs, self.action_dim, 2)
                index_list, _, value_list = string_utils.resolve_matching_names_values(self.cfg.clip, self._joint_names)
                for idx, (min_val, max_val) in zip(index_list, value_list):
                    self._clip[:, idx] =
            else:
                raise ValueError(f"Unsupported clip type: {type(cfg.clip)}. Supported types are dict.")
```

I don't know how to put the value into `self._clip[:, idx]`, direct assignment will cause broadcasting problems

Reply:
solved.